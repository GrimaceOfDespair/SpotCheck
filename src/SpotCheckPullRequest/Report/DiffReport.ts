export interface IDiffTestRun {
    total: number,
    totalPassed: number,
    totalFailed: number,
    totalSuites: number,
    suites: IDiffSuite[]
}

export interface IDiffSuite {
    name: string;
    path: string;
    tests: IDiffTest[];
}

export interface IDiffTest {
    name: string;
    specPath: string;
    specFilename: string;
    status: 'pass' | 'fail';
    percentage: number;
    failureThreshold: number;
    baselinePath: string;
    comparisonPath: string;
    diffPath: string;
}
