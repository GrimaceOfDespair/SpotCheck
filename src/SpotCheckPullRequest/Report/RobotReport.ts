export interface IRobotSuite {
    name: string;
    path: string;
    tests: IRobotTest[];
}

export interface IRobotTest {
    test: string;
    suite: string;
    source: string;
    imageName: string;
    imageThreshold: string;
    imageFolder: string;
}

export interface IRobotSuites {
    [suite: string]: IRobotTest[];
}