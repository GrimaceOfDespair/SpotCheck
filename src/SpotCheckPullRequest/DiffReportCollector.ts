import { IDiffTestReport } from "./Report/DiffReport";
import { Temp } from "./Temp";
import fs from 'node:fs';
import path from 'node:path';

export class DiffReportCollector {

    private _reportName: string;
    private _screenshots: string;

    constructor(reportName?: string, screenshots?: string) {
        this._reportName = reportName ?? 'diff-report.json';
        this._screenshots = screenshots ?? '';
    }

    async collectReportFile(reportPath: string): Promise<string> {
        const output = await Temp.createFolder();
        const targetReport = path.join(output, this._reportName);
        await fs.promises.copyFile(reportPath, targetReport);

        const reportData = await fs.promises.readFile(reportPath);
        const report = <IDiffTestReport>JSON.parse(reportData.toString());

        const baseDir = path.dirname(reportPath);
        await this._collectScreenshots(baseDir, output, report);

        return output;
    }

    async collectReport(baseDir: string, report: IDiffTestReport): Promise<string> {
        const output = await Temp.createFolder();

        await this._collectScreenshots(baseDir, output, report);

        return output;
    }

    private async _collectScreenshots(source: string, target:string, report: IDiffTestReport) {
        for (const test of report.suites.flatMap(s => s.tests)) {
            await this._copyImage(source, target, test.baselinePath);
            await this._copyImage(source, target, test.comparisonPath);
            await this._copyImage(source, target, test.diffPath);
        }
    }

    private async _copyImage(source: string, target: string, image: string) {
        if (!image) {
            return;
        }

        const sourceImage = path.join(source, this._screenshots, image);
        const targetImage = path.join(target, this._screenshots, image);
        const targetFolder = path.dirname(targetImage);

        await fs.promises.mkdir(targetFolder, { recursive: true });
        await fs.promises.copyFile(sourceImage, targetImage,);
    }
}