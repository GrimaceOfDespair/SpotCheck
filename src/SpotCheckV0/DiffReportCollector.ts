import { IDiffTestReport } from "./Report/DiffReport";
import { ILogger, NullLogger } from "./Report/ILogger";
import { Temp } from "./Temp";
import fs from 'node:fs';
import path from 'node:path';

export class DiffReportCollector {

    private _reportName: string;
    private _screenshots: string;
    private _logger: ILogger;

    constructor(reportName?: string, screenshots?: string, logger?: ILogger) {
        this._reportName = reportName ?? 'diff-report.json';
        this._screenshots = screenshots ?? '';
        this._logger = logger ?? NullLogger;
    }

    async collectReport(baseDir: string, report: IDiffTestReport): Promise<string> {
        const output = await Temp.createFolder();
        const targetReport = path.join(output, this._reportName);
        
        this._logger.info(`Writing report to ${targetReport}`);

        await fs.promises.writeFile(targetReport, JSON.stringify(report));

        await this._collectScreenshots(baseDir, output, report);

        return output;
    }

    private async _collectScreenshots(source: string, target:string, report: IDiffTestReport) {

        this._logger.info(`Collect screenshots from ${source} into ${target}`);

        for (const test of report.suites.flatMap(s => s.tests)) {
            await this._copyImage(source, target, test.baselinePath);
            await this._copyImage(source, target, test.comparisonPath);
            await this._copyImage(source, target, test.diffPath);
        }
    }

    private async _copyImage(source: string, target: string, image: string) {
        if (!image) {
            return;
        }

        const sourceImage = path.join(source, this._screenshots, image);
        const targetImage = path.join(target, this._screenshots, image);
        const targetFolder = path.dirname(targetImage);

        await fs.promises.mkdir(targetFolder, { recursive: true });
        await fs.promises.copyFile(sourceImage, targetImage,);
    }
}