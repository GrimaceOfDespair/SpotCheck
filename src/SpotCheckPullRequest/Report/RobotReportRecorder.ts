import { Tag } from "sax";
import { ITestContext, TestStatus } from "../TestContext";
import { IRobotSuites, IRobotTest } from "./RobotReport";
import { SaxParserRecorder, State } from "./SaxParserRecorder";

export class RobotReportRecorder implements SaxParserRecorder {

    tests: IRobotTest[] = [];
    isKeyword: boolean = false;
    isArg: boolean = false;
    isStatus: boolean = false;
    captureScreenshot: string = '';
    plainText: boolean = false;
    testContext: ITestContext = {
        suiteStack: [],
        args: [],
        source: '',
        test: '',
        status: 'NOT RUN',
    };

    getGroupedSuites(): IRobotSuites {
        return this.tests.reduce((suites, test) => {
            (suites[test.suite] = suites[test.suite] || []).push(test);
            return suites;
        }, <IRobotSuites>{});
    }

    normalize(screenshot: string) {
        const normalizedScreenshot = screenshot
            .replace(/ /g, '_')
            .replace(/[^-\w.]/g, '');

        switch (normalizedScreenshot) {
            case '':
            case '.':
            case '..':
                throw new Error(`Cannot create file name from ${screenshot} as it results in ${normalizedScreenshot}`);

            default:
                return normalizedScreenshot;
        }
    }

    start(_: State) {
    }

    stop(_: State) {
        if (!this.testContext.test || this.testContext.status != 'PASS') {
            return;
        }

        const { suiteStack, test, source } = this.testContext;
        const [ snapshot, imageThreshold ] = this.testContext.args;
        const suite = suiteStack[suiteStack.length-1];
        const imageName = this.normalize([...suiteStack, snapshot, 'png'].join('.'));

        this.tests.push({
            suite,
            test,
            source,
            imageName,
            imageThreshold,
        });

        this.testContext.args = [];
    }

    onOpenTag(node: Tag) {
        switch (node.name)
        {
            case 'suite':
                this.testContext.suiteStack.push(node.attributes['name']);
                this.testContext.source = node.attributes['source'];
                break;

            case 'test':
                this.testContext.test = node.attributes['name'];
                break;

            case 'kw':
                this.isKeyword = true;
                this.testContext.args = [];
                break;

            case 'arg':
                this.isArg = true;
                break;

            case 'status':
                this.testContext.status = <TestStatus>node.attributes['status'];
                break;
        }
    }

    onCloseTag(tag: string) {
        
        switch (tag) {
            case 'suite':
                this.testContext.suiteStack.pop();
                break;

            case 'kw':
                this.isKeyword = false;
                break;

            case 'arg':
                this.isArg = false;
                break;
        }
    }

    onText(text: string) {
        if (this.isKeyword && this.isArg) {
            this.testContext.args.push(text);
        }
    }

    onCDATA(cdata: string) {
    }

    onOpenCDATA(tag: string) {
    }

    onCloseCDATA(tag: string) {
    }
}