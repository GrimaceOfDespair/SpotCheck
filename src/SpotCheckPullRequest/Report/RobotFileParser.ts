import { createReadStream, writeFile } from "node:fs";
import path from 'node:path';
import sax from 'sax';
const { SaXPath } = require('saxpath');
//import { SaXPath } from 'saxpath';
import { once } from 'events';
import { IRobotSuite, IRobotTest, IRobotSuites } from "./RobotReport";
import { ReportContext } from "../ReportContext";
import { IDiffSuite, IDiffTest, IDiffTestRun } from "./DiffReport";
import { ImageProcessor } from "../ImageProcessor";
import { RobotReportRecorder } from "./RobotReportRecorder";

export class RobotFileParser {

    defaultThreshold: number = .05;
    
    private _context: ReportContext;
    private _images: ImageProcessor = new ImageProcessor();

    constructor(reportFile: string, baseDir?: string) {
        this._context = new ReportContext(reportFile, baseDir ?? path.dirname(reportFile));
    }

    async parseReport(imageFolder: string): Promise<IDiffTestRun> {

        const testSuites = await this.readTestSuites();
        
        return await this.parseScreenshots(testSuites, imageFolder);
    }

    private async parseScreenshots(groupedSuites: IRobotSuites, imageFolder: string): Promise<IDiffTestRun> {

        const suites: IDiffSuite[] = await Promise.all(Object.entries(groupedSuites)
            .map(async ([suite, tests]) => ({
                name: suite,
                path: tests[0].source,
                tests: await Promise.all(tests.map(async (test) => ({
                    name: test.test,
                    specPath: test.source,
                    specFilename: path.basename(test.source),
                    ...await this.getScreenshot(test.imageName, test.imageThreshold, imageFolder)
                })))
            })));

        const allTests = suites.reduce((acc, suite) => acc.concat(suite.tests), <IDiffTest[]>[]);

        return {
            total: allTests.length,
            totalPassed: allTests.filter(test => test.status == 'pass').length,
            totalFailed: allTests.filter(test => test.status == 'fail').length,
            totalSuites: suites.length,
            suites
        };
    }

    private async readTestSuites() {

        const reportStream = createReadStream(this._context.reportFile);
        const saxParser = sax.createStream(true);

        const testContext = {
            suite: [],
            test: '',
            keyword: '',
        };

        const suiteRecorder = new RobotReportRecorder();
        const saxPath = new SaXPath(saxParser, `//test//kw[@name="Compare Snapshot"]`, suiteRecorder);
        reportStream.pipe(saxParser);

        await once(saxPath, 'end');

        let groupedSuites = suiteRecorder.getGroupedSuites();
        return groupedSuites;
    }

    async getScreenshot(imageName: string, imageThreshold: string, imageFolder: string):
        Promise<Omit<IDiffTest, "name" | "specPath" | "specFilename">> {

        const imageNameOnDisk = decodeURIComponent(imageName);
        const imageFolderOnDisk = decodeURIComponent(imageFolder);
        const relativeFolder = this._context.baseDir
            ? imageFolderOnDisk.replace(new RegExp('^' + this._context.baseDir + '/?'), '')
            : imageFolderOnDisk;
    
        const [baselinePath, comparisonPath, diffPath] =
            ['baseline', '', 'diff'].map(version =>
            ({
                relative: [relativeFolder, version, imageNameOnDisk].filter(Boolean).join('/'),
                absolute: [this._context.basePath, imageFolderOnDisk, version, imageNameOnDisk].filter(Boolean).join('/')
            }));
    
        let failureThreshold = parseFloat(imageThreshold);
        if (isNaN(failureThreshold)) {
            failureThreshold = this.defaultThreshold;
        }
    
        const { percentage, testFailed } = await this._images.compareImage(
            baselinePath.absolute,
            comparisonPath.absolute,
            diffPath.absolute,
            failureThreshold);
    
        return {
            status: testFailed ? 'fail' : 'pass',
            percentage,
            failureThreshold,
            baselinePath: baselinePath.relative,
            comparisonPath: comparisonPath.relative,
            diffPath: testFailed ? diffPath.relative : ''
        };
    }}