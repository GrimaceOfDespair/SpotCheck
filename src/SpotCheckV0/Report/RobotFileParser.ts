import { createReadStream, writeFile } from "node:fs";
import path from 'node:path';
import sax from 'sax';
const { SaXPath } = require('saxpath');
//import { SaXPath } from 'saxpath';
import { once } from 'events';
import { IRobotSuite, IRobotTest, IRobotSuites } from "./RobotReport";
import { ReportContext } from "../ReportContext";
import { IDiffSuite, IDiffTest, IDiffTestReport } from "./DiffReport";
import { ImageProcessor } from "../ImageProcessor";
import { RobotReportRecorder } from "./RobotReportRecorder";
import { ILogger, NullLogger } from "./ILogger";

export class RobotFileParser {

    defaultThreshold: number = .05;

    private _context: ReportContext;
    private _images: ImageProcessor = new ImageProcessor();
    private _logger: ILogger;

    constructor(reportFile: string, baseDir?: string, logger?: ILogger) {
        this._context = new ReportContext(reportFile,
            baseDir ?? path.dirname(reportFile));
        this._logger = logger ?? NullLogger;
    }

    async createDiffReport(normalizePaths: boolean = true): Promise<IDiffTestReport> {

        const testSuites = await this.readTestSuites(normalizePaths);
        
        return await this.parseScreenshots(testSuites);
    }

    private async parseScreenshots(groupedSuites: IRobotSuites): Promise<IDiffTestReport> {

        const suites: IDiffSuite[] = await Promise.all(Object.entries(groupedSuites)
            .map(async ([suite, tests]) => ({
                name: suite,
                path: tests[0].source,
                tests: await Promise.all(
                    tests.map(async ({ test: name, source: specPath, imageName, imageThreshold }) => ({
                        name,
                        specPath,
                        specFilename: path.basename(specPath),
                        ...await this.getScreenshot(imageName, imageThreshold)
                    })))
            })));

        const allTests = suites.reduce((acc, suite) => acc.concat(suite.tests), <IDiffTest[]>[]);

        return {
            total: allTests.length,
            totalPassed: allTests.filter(test => test.status == 'pass').length,
            totalFailed: allTests.filter(test => test.status == 'fail').length,
            totalSuites: suites.length,
            suites
        };
    }

    private async readTestSuites(normalizePaths: boolean) {

        const reportStream = createReadStream(this._context.reportFile);
        const saxParser = sax.createStream(true);

        const suiteRecorder = new RobotReportRecorder(normalizePaths);
        const saxPath = new SaXPath(saxParser, `//test//kw[@name="Compare Snapshot"]`, suiteRecorder);
        reportStream.pipe(saxParser);

        await once(saxPath, 'end');

        let groupedSuites = suiteRecorder.getGroupedSuites();
        return groupedSuites;
    }

    async getScreenshot(imageName: string, imageThreshold: string):
        Promise<Omit<IDiffTest, "name" | "specPath" | "specFilename">> {

        const imageNameOnDisk = decodeURIComponent(imageName);
    
        const [baselinePath, comparisonPath, diffPath] =
            ['baseline', '', 'diff'].map(version =>
                ({
                    relative: this._context.resolveRelative(version, imageNameOnDisk),
                    absolute: this._context.resolveAbsolute(version, imageNameOnDisk),
                })
            );
    
        let failureThreshold = parseFloat(imageThreshold);
        if (isNaN(failureThreshold)) {
            failureThreshold = this.defaultThreshold;
        }

        this._logger.info(`Comparing '${baselinePath.absolute}' with '${comparisonPath.absolute}' with threshold ${Math.round(failureThreshold * 100)}% and writing output to ${diffPath.absolute}`);
    
        const { percentage, testFailed } = await this._images.compareImage(
            baselinePath.absolute,
            comparisonPath.absolute,
            diffPath.absolute,
            failureThreshold);

        if (testFailed) {
            this._logger.error(`Failed with ${Math.round(percentage * 100)}% difference`);
        } else {
            this._logger.info(`Passed with ${Math.round(percentage * 100)}% difference`);
        }
    
        return {
            status: testFailed ? 'fail' : 'pass',
            percentage,
            failureThreshold,
            baselinePath: baselinePath.relative,
            comparisonPath: comparisonPath.relative,
            diffPath: testFailed ? diffPath.relative : ''
        };
    }}