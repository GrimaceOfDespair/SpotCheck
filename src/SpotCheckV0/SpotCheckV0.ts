import tl = require('azure-pipelines-task-lib/task');
import { RobotFileParser } from './Report/RobotFileParser';
import path from 'node:path';
import fs from 'node:fs';
import { DiffReportCollector } from './DiffReportCollector';
import { PullRequestHandler } from './PullRequestHandler';
import { IDiffTestReport } from './Report/DiffReport';
import { ILogger } from './Report/ILogger';

function throwExpression(errorMessage: string): never {
    throw new Error(errorMessage);
}

 (async function() {
     try {
        const [ input, output, artifactName, screenshotFolder, type, baseDir ] = [
            [ 'input', '', 'specify a report file for "input"' ],
            [ 'output', 'diff-report.json' ],
            [ 'artifactName', 'screenshots' ],
            [ 'screenshotFolder', '' ],
            [ 'type', 'robot' ],
            [ 'baseDir', path.dirname(tl.getInput('input', false) ?? '') ],
        ]
        .map(([ key, defaultValue, errorMessage ]) =>
            tl.getInput(key, errorMessage !== undefined)
                ?? (errorMessage !== undefined
                    ? throwExpression(errorMessage)
                    : defaultValue));

        tl.debug(`type: ${type}`);
        tl.debug(`baseDir: ${baseDir}`);
        tl.debug(`output: ${output}`);
        tl.debug(`screenshotFolder: ${screenshotFolder}`);

        const skipFeedback = tl.getBoolInput('skipFeedback', false);
        const taskLogger: ILogger = skipFeedback ? {
            error: (_) => {},
            info: (_) => {}
        } : {
            error: (message) => tl.warning(message),
            info: (message) => tl.debug(message)
        };

        let diffReport: IDiffTestReport;
        switch (type) {
            case 'robot':

                diffReport = await new RobotFileParser(input, baseDir, taskLogger)
                    .createDiffReport();

                break;

            case 'cypress':
                const reportData = await fs.promises.readFile(input);
                diffReport = <IDiffTestReport>JSON.parse(reportData.toString());

                break;

            default:
                throw new Error(`"type" should be "robot" or "cypress", but was "${type}"`);
        }
        
        const diffReportCollector = new DiffReportCollector(output, screenshotFolder, taskLogger);

        const outputPath = await diffReportCollector.collectReport(baseDir, diffReport);
        tl.uploadArtifact('', outputPath, artifactName);

        if (!skipFeedback) {
            const prHandler = new PullRequestHandler(tl);
            prHandler.processFeedback(diffReport);
        }
     }
     catch (err:any) {
         tl.setResult(tl.TaskResult.Failed, err.message);
     }
 })();