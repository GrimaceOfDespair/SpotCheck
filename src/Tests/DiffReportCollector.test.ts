import { DiffReportCollector } from "../SpotCheckV0/DiffReportCollector";
import path from 'node:path';
import fs from 'node:fs';
import { RobotFileParser } from "../SpotCheckV0/Report/RobotFileParser";
import { IDiffTestReport } from "../SpotCheckV0/Report/DiffReport";

describe('DiffReportCollector', () => {

    test('Collect simple report with two passing screenshots', async () => {

        // Arrange
        const report = <IDiffTestReport>require('./reports/output-pass.json');
        const baseDir = path.join(__dirname, 'reports');

        // Act
        const collectedReportFolder = await new DiffReportCollector('output.json', 'screenshots')
            .collectReport(baseDir, report);

        // Assert
        const imageReport = path.join(collectedReportFolder, 'output.json');
        const imageReportJson = await fs.promises.readFile(imageReport, { encoding: 'utf-8' });
        
        expect(imageReportJson).toMatchSnapshot();

        const collectedImageFolder = path.join(collectedReportFolder, 'screenshots');
        const collectedImages = (await fs.promises.readdir(collectedImageFolder)).map(image =>
            path.basename(image));

        expect(collectedImages.sort()).toEqual([
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png',
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png',
            'baseline']);
    });

    test('Collect simple report with default path', async () => {

        // Arrange
        const report = <IDiffTestReport>require('./reports/output-pass.json');
        const baseDir = path.join(__dirname, 'reports', 'screenshots');

        // Act
        const collectedReportFolder = await new DiffReportCollector()
            .collectReport(baseDir, report);

        // Assert
        const imageReport = path.join(collectedReportFolder, 'diff-report.json');
        const imageReportJson = await fs.promises.readFile(imageReport, { encoding: 'utf-8' });

        expect(imageReportJson).toMatchSnapshot();

        const collectedImages = (await fs.promises.readdir(collectedReportFolder)).map(image =>
            path.basename(image));

        expect(collectedImages.sort()).toEqual([
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png',
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png',
            'baseline',
            'diff-report.json']);
    });

    test('Collect simple report without screenshot folder', async () => {

        // Arrange
        const report = <IDiffTestReport>require('./reports/output-subfolder.json');
        const baseDir = path.join(__dirname, 'reports');

        // Act
        const collectedReportFolder = await new DiffReportCollector('output.json')
            .collectReport(baseDir, report);

        // Assert
        const imageReport = path.join(collectedReportFolder, 'output.json');
        const imageReportJson = await fs.promises.readFile(imageReport, { encoding: 'utf-8' });
        
        expect(imageReportJson).toMatchSnapshot();

        const collectedImageFolder = path.join(collectedReportFolder, 'screenshots');
        const collectedImages = (await fs.promises.readdir(collectedImageFolder)).map(image =>
            path.basename(image));

        expect(collectedImages.sort()).toEqual([
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png',
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png',
            'baseline']);
    });

    test('Collect simple report with passing and failing screenshot', async () => {

        // Arrange
        const report = <IDiffTestReport>require('./reports/output-fail.json');
        const baseDir = path.join(__dirname, 'reports');

        // Act
        const collectedReportFolder = await new DiffReportCollector('output.json', 'screenshots')
            .collectReport(baseDir, report);

        // Assert
        const imageReport = path.join(collectedReportFolder, 'output.json');
        const imageReportJson = await fs.promises.readFile(imageReport, { encoding: 'utf-8' });
        
        expect(imageReportJson).toMatchSnapshot();

        const collectedImageFolder = path.join(collectedReportFolder, 'screenshots');
        const collectedImages = (await fs.promises.readdir(collectedImageFolder)).map(image =>
            path.basename(image));

        expect(collectedImages.sort()).toEqual([
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png',
            'Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png',
            'baseline']);
    });
})