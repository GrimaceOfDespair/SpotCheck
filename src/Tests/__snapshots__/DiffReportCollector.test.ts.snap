// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffReportCollector Collect simple report with default path 1`] = `"{"total":2,"totalPassed":0,"totalFailed":2,"totalSuites":1,"suites":[{"name":"Listsandstaticsegments","path":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","tests":[{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.1,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","diffPath":""},{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.05,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","diffPath":""}]}]}"`;

exports[`DiffReportCollector Collect simple report with passing and failing screenshot 1`] = `"{"total":2,"totalPassed":0,"totalFailed":2,"totalSuites":1,"suites":[{"name":"Listsandstaticsegments","path":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","tests":[{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.1,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","diffPath":""},{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"fail","percentage":1,"failureThreshold":0.05,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","diffPath":""}]}]}"`;

exports[`DiffReportCollector Collect simple report with two passing screenshots 1`] = `"{"total":2,"totalPassed":0,"totalFailed":2,"totalSuites":1,"suites":[{"name":"Listsandstaticsegments","path":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","tests":[{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.1,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","diffPath":""},{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.05,"baselinePath":"baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","comparisonPath":"Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","diffPath":""}]}]}"`;

exports[`DiffReportCollector Collect simple report without screenshot folder 1`] = `"{"total":2,"totalPassed":0,"totalFailed":2,"totalSuites":1,"suites":[{"name":"Listsandstaticsegments","path":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","tests":[{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.1,"baselinePath":"screenshots/baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","comparisonPath":"screenshots/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard.png","diffPath":""},{"name":"Create_List","specPath":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","specFilename":"X:\\\\Tests\\\\Automation\\\\test_suites\\\\89_sanitychecks\\\\listsandstaticsegments.robot","status":"pass","percentage":1,"failureThreshold":0.05,"baselinePath":"screenshots/baseline/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","comparisonPath":"screenshots/Test_Suites.89_Sanitychecks.Listsandstaticsegments.Lists_Dashboard_Without_Threshold.png","diffPath":""}]}]}"`;
